src/background/stores/assessment-card-selection-store.ts

```diff
@@ -221,4 +282,134 @@ export class AssessmentCardSelectionStore extends PersistentStore<AssessmentCard
 
         await this.emitChanged();
     };
+
+    private onScanCompleted = async (payload: ScanCompletedPayload<any>) => {
+        if (
+            !payload ||
```

Comment:
Is there a reason we are validating so much of the payload? We don't usually do that ...